auth/client/src/main/java/org/wildfly/security/auth/client/ElytronClientDefaultSSLContextProvider.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2021 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.auth.client;
+
+import org.wildfly.client.config.ConfigXMLParseException;
+import org.wildfly.security.auth.client._private.ElytronMessages;
+
+import java.net.URISyntaxException;
+import java.security.GeneralSecurityException;
+import java.security.NoSuchAlgorithmException;
+import java.security.Provider;
+import java.util.List;
+import java.util.Map;
+
+public final class ElytronClientDefaultSSLContextProvider extends Provider {
+
+    public ElytronClientDefaultSSLContextProvider() {
+        this(null);
+    }
+
+    public ElytronClientDefaultSSLContextProvider(String configPath) {
+        super("ElytronClientDefaultSSLContextProvider", 1.0, "Elytron client provider for default SSLContext");
```

Comment:
@fabiobrz The `super(String, String, String)` constructor was introduced in JDK 9, so I canot use it here because we have release configured for compiler to be 8. The deprecated constructor I used is still present in JDK 15 so I used it but good point, I will check today with the team whether the release can be changed to 9. If yes I can change it, if not I have to leave it as it is. I will let you know

Reply:
Thank you very much @Skyllarr - not a big issue, though. There is valid reasoning already for leaving it as is, which you provided.

I'll leave the final call to you here.

Reply:
@fabiobrz Yes we have to use the deprecated constructor here, as wildfly should work with JDK 8.